### PR TITLE
CESM-GC updates for cam_development

### PR DESCRIPTION
**Pull request by lizziel**
This is a dummy PR to show documentation of CESM-GC changes to merge into the main CAM development branch. The source branch is a version of CESM-GC_main rebased onto CAM 6.0.034, with all CAM code in stale CESM 2.1.1 branch dropped, and many CESM-GC commits squashed.

The latest cam_development (target) has not been merged into the source branch and there are thus conflicts. This dummy PR also serves to help sort out those conflicts.

### CODE REVIEW

**Comment by lizziel** [comment]
@fritzt and @jimmielin, this dummy PR is a precursor to making a PR on the upstream CAM and discussing changes with NCAR. Could you look over the file changes and tell me if anything should be excluded from the PR, or if anything looks wrong?

A bit of information on what you are looking at:
The source branch is a modified version of CESM-GC_main. I squashed 223 commits from the three of us into 12 commits total, and then dropped CESM commits between CAM 6.0.034 and the CESM 2.1.1 version Thibaud started with. I then merged the latest cam_development as of yesterday into it and did conflict resolution. The only features I have removed/disabled so far from the original changes are (1) change HEMCO from on to off by default (thanks for the very clear message about that in the code!), and (2) remove AEIC species from the external forcing list.

I expect that I should change more before opening the ESCOMP PR and I'm going to look at the files now. Let's discuss on this thread.

File: src/chemistry/pp_trop_strat_mam4_vbs/mo_sim_dat.F90 [hunk 1]
```diff
@@ -246,14 +246,14 @@ subroutine set_sim_dat
                           1278,1291,1304,1327,1352,1507,1549,1640,1691,1716, &
                           1739,1845,1876,1900,1935,1993,2054,2080 /)
 
-      extfrc_lst(: 16) = (/ 'so4_a1          ','bc_a4           ','SVOC            ','bc_a1           ','CO              ', &
+      extfrc_lst(: 24) = (/ 'so4_a1          ','bc_a4           ','SVOC            ','bc_a1           ','CO              ', &
```
Comment (fritzt) [inline src/chemistry/pp_trop_strat_mam4_vbs/mo_sim_dat.F90]: The indexing is wrong here. There are only 16 elements. Similar indexing errors also appear below and in other `mo_sim_dat.F90`

File: src/chemistry/geoschem/gas_wetdep_opts.F90 [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    do i = 1,pcnst
+       if ( len_trim(gas_wetdep_list(i)) > 0 ) then
+          gas_wetdep_cnt = gas_wetdep_cnt + 1
+       endif
+    enddo
+
+    if (( gas_wetdep_cnt>0 ).and. &
+       ( .not.(gas_wetdep_method=='MOZ' .or. &
+               gas_wetdep_method=='NEU' .or. &
+               gas_wetdep_method=='GEOS-CHEM' .or. &
```
Comment (fritzt) [inline src/chemistry/geoschem/gas_wetdep_opts.F90]: Just realized this now, but this line "`gas_wetdep_method=='GEOS-CHEM' .or. &`" can be removed. We used to have the option to perform scavenging using the GEOS-Chem approach, but this doesn't work out in CESM2-GC.

File: src/chemistry/pp_trop_strat_mam4_vbs/mo_sim_dat.F90 (hunk 1 above)
Comment (lizziel) [reply src/chemistry/pp_trop_strat_mam4_vbs/mo_sim_dat.F90]: Thanks for catching that. I manually reverted the changes and forgot that index.

File: src/chemistry/geoschem/gas_wetdep_opts.F90 (hunk 2 above)
Comment (lizziel) [reply src/chemistry/geoschem/gas_wetdep_opts.F90]: I'll remove it.

**Comment by jimmielin** [comment]
Hi Lizzie,

Thanks so much for working on this. I'm mostly familiar with the HEMCO changes. This all looks good to me. Disabling HEMCO by default is good, for it to work with CAM-chem we still have to work with the CAM-chem folks to add the changes to their emissions code (the changes in https://github.com/jimmielin/HEMCO_CESM/tree/development/SourceMods). We can sort that out in the future.

Best,
Haipeng

**Review by jimmielin — APPROVED** [review]

**Comment by fritzt** [comment]
Hi Lizzie! Thank you for putting all of this together! This looks good to me. Some of the files were copied over from the `mozart` folder and are mostly commented out (e.g. `charge_neutrality.F90`, `chem_prod_loss_diags.F90`, `clybry_fam.F90`, `mo_gas_phase_chemdr.F90`, ...). I'm wondering if it would be worth it to remove all of this commented code. However, we can't just delete them, as some of these files have dependencies with other CAM modules.

**Review by fritzt — APPROVED** [review]

**Comment by lizziel** [comment]
See new PR at https://github.com/ESCOMP/CAM/pull/484

**Pull request closed without merging**